blocks/rich-text/index.js

```diff
@@ -558,6 +559,44 @@ export class RichText extends Component {
 		if ( keyCode === BACKSPACE ) {
 			this.onChange();
 		}
+
+		// `scrollToCaret` is called on `nodechange`, whereas calling it on
+		// `keyup` *when* moving to a new RichText element results in incorrect
+		// scrolling. Though the following allows false positives, it results
+		// in much smoother scrolling.
+		if ( keyCode !== BACKSPACE && keyCode !== ENTER ) {
+			this.scrollToCaret();
+		}
+	}
+
+	scrollToCaret() {
+		if ( ! this.props.isViewportSmall ) {
+			return;
+		}
+
+		const { top: caretTop } = this.getEditorSelectionRect();
+
+		const container = getScrollContainer( this.editor.getBody() );
```

Comment:
Could this be cached?

Reply:
I'd say yes, but what would invalidate it? Resizing? Viewport breakpoints? Or do we assume the container to be constant throughout RichText's lifetime?

Reply:
Oh, right, I somehow thought it would remain constant.